server/swap/swap.go

```diff
@@ -2287,22 +2290,6 @@ func (s *Swapper) processMatchAcks(user account.AccountID, msg *msgjson.Message,
 	// Remove the live acker from Swapper's tracking.
 	defer s.rmLiveAckers(msg.ID)
 
-	// Lock each matchTracker as early as possible to try to avoid 'init'
-	// requests for these matches failing because the ack signatures have not
-	// yet been stored. A better solution would be a response to the client so
-	// they know when it's OK to 'init', but we presently lack a response to a
-	// response mechanism.
-	uniqueTrackers := make(map[*matchTracker]bool, len(matches)/2)
-	for _, matchInfo := range matches {
-		mt := matchInfo.match
-		if uniqueTrackers[mt] {
-			continue
-		}
-		uniqueTrackers[mt] = true
-		mt.mtx.Lock()
-		defer mt.mtx.Unlock()
```

Comment:
If there were two concurrent match acks targeting the same match ID, this would deadlock in the loop.  Really a poor idea in retrospect.